rtl/eclrtl/rtlnewkey.cpp

```diff
@@ -0,0 +1,2133 @@
+        StringBuffer upperString, lowerString;
+        if (*filter=='\'')
+        {
+            filter++;
+            readString(lowerString, filter);
+        }
+        else
+            readUntilTerminator(lowerString, filter, ",])");
+
+        if (*filter == ',')
+        {
+            filter++;
+            if (*filter=='\'')
+            {
+                filter++;
+                readString(upperString, filter);
+            }
+            else
+                readUntilTerminator(upperString, filter, "])");
+        }
+        else
+            upperString.set(lowerString);
+
+        char endRange = *filter++;
+        if (endRange != ')' && endRange != ']')
+            throw MakeStringException(0, "Invalid filter string: expected ] or ) at end of range");
+        if (*filter==',')
+            filter++;
+        else if (*filter)
+            throw MakeStringException(0, "Invalid filter string: expected , between ranges");
+
+        TransitionMask lowerMask = (startRange == '(') ? CMPgt : CMPge;
+        TransitionMask upperMask = (endRange == ')') ? CMPlt : CMPle;
+        creator->addRange(lowerMask, lowerString, upperMask, upperString);
+    }
+}
+
+class ValueSetCreator : implements ISetCreator
+{
+public:
+    ValueSetCreator(IValueSet * _set) : set(_set) {}
+
+    virtual void addRange(TransitionMask lowerMask, const StringBuffer & lowerString, TransitionMask upperMask, const StringBuffer & upperString) override
+    {
+        Owned<ValueTransition> lower = lowerString ? set->createUtf8Transition(lowerMask, rtlUtf8Length(lowerString.length(), lowerString), lowerString) : nullptr;
+        Owned<ValueTransition> upper = upperString ? set->createUtf8Transition(upperMask, rtlUtf8Length(upperString.length(), upperString), upperString) : nullptr;
+        set->addRange(lower, upper);
+    }
+
+protected:
+    IValueSet * set;
+};
+
+
+
+void deserializeSet(IValueSet * set, const char * filter)
+{
+    ValueSetCreator creator(set);
+    deserializeSet(&creator, filter);
+}
+
+//---------------------------------------------------------------------------------------------------------------------
+
+//This class represents a value and a comnparison condition and is used for representing ranges of values
+class ValueTransition : implements CInterface
+{
+public:
+    ValueTransition(TransitionMask _mask, const RtlTypeInfo & type, const void *_value)
+    {
+        mask = _mask;
+        dbgassertex(_value || isMinimum() || isMaximum());
+        if (_value)
+        {
+            size32_t size = type.size((const byte *)_value, nullptr);
+            value.allocateN(size, false);
+            memcpy(value, _value, size);
+        }
+    }
+    ValueTransition(TransitionMask _mask) : mask(_mask)
+    {
+        dbgassertex(isMaximum() || isMinimum());
+    }
+
+    MemoryBuffer &serialize(const RtlTypeInfo & type, MemoryBuffer &mb) const
+    {
+        mb.append(mask);
```

Comment:
This gives me a compile error:

/Users/rchapman/HPCC-Platform/rtl/eclrtl/rtlnewkey.cpp:156:12: error: call to member function 'append' is ambiguous